Mathlib/Algebra/Group/EvenFunction.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Algebra.Group.Action.Pi
+import Mathlib.Algebra.Module.Defs
+
+/-!
+# Even and odd functions
+
+We define even functions `α → β` assuming `α` has a negation, and odd functions assuming both `α`
+and `β` have negation.
+
+These definitions are called `EvenFun` and `OddFun` to avoid conflicting with the root-level
+definitions `Even` and `Odd` (which, for functions, mean that the function takes even resp. odd
+_values_, a wholly different concept).
+-/
+
+namespace Function
+
+variable {α β : Type*} [Neg α]
+
+/-- A function `f` is _even_ if it satisfies `f (-x) = f x` for all `x`. -/
+def EvenFun (f : α → β) : Prop := ∀ a, f (-a) = f a
```

Comment:
I know it was discussed a bit on Zulip, but my preference here would be for:
```suggestion
def protected Function.Even (f : α → β) : Prop := ∀ a, f (-a) = f a
```

Reply:
The Zulip discussion didn't have a big majority for either option, and on consideration I actually agree with Jireh that `Function.Even` is better, so I'll change it.